wodles/aws/aws_s3.py

```diff
@@ -3411,47 +3577,35 @@ class AWSSQSQueue(WazuhIntegration):
         URL for the VPC endpoint to use to obtain the STS token.
     service_endpoint : str
         URL for the endpoint to use to obtain the logs.
+    message_processor: AWSQueueMessageProcessor
+        Class to process received notifications.
     """
 
-    def __init__(self, name: str, iam_role_arn: str, access_key: str = None, secret_key: str = None,
-                 external_id: str = None, sts_endpoint=None, service_endpoint=None, **kwargs):
-        self._validate_params(external_id=external_id, name=name, iam_role_arn=iam_role_arn)
+    def __init__(self, name: str, iam_role_arn: str, message_processor: AWSQueueMessageProcessor,
+                 bucket_handler: AWSS3LogHandler,
```

Comment:
This is like an abstract class, why you are passing as a parameter and instantiating below?

Reply:
My bad here is used as a type hint.